end_to_end/test_checkpointing.sh

```diff
@@ -11,6 +11,10 @@ python3 MaxText/train.py MaxText/configs/base.yml run_name=$RUN_NAME steps=501\
     metrics_file='saved_metrics.txt' save_period=20 base_output_directory=$OUTPUT_PATH dataset_path=$DATASET_PATH\
     collect_stack_trace=$COLLECT_STACK_TRACE
 
+# Wait for first train to finish
+process_id=$!
+wait $process_id
```

Comment:
What is the need to add a wait here?

Reply:
There was some issues with that in the past. I added it just as a safety net. It was approved here: #76 , I just forgot to rebase for this PR.